src/compiler/cpp_generator.cc

```diff
@@ -1574,11 +1589,25 @@ void PrintSourceService(grpc_generator::Printer* printer,
   (*vars)["Service"] = service->name();
 
   if (service->method_count() > 0) {
+    std::vector<size_t> method_name_lens;
```

Comment:
nit: I know it probably doesn't matter, but might be worth adding `reserve(service->method_count())`